lib/metadata-json-lint/schema.rb

```diff
@@ -0,0 +1,150 @@
+  # maintained in this class. Provides a good first pass over the metadata with
+  # type checking and basic format/length validations.
+  #
+  # rubocop:disable Metrics/ClassLength # schema data structure is quite large
+  class Schema
+    # Based on https://docs.puppet.com/puppet/latest/modules_metadata.html
+    #
+    # rubocop:disable Style/TrailingCommaInLiteral # easier to modify individual lines
+    def schema
+      {
+        '$schema' => 'http://json-schema.org/draft-04/schema#',
+        'properties' => {
+          'author' => {
+            'type' => 'string',
+          },
+          'data_provider' => {
+            'type' => %w[null string],
+          },
+          'dependencies' => {
+            'type' => 'array',
+            'items' => {
+              'properties' => {
+                'name' => {
+                  'type' => 'string',
+                  'pattern' => '^\w+[/-][a-z][a-z0-9_]*$',
+                },
+                'version_requirement' => {
+                  'type' => 'string',
+                },
+              },
+              'required' => %w[
+                name
+              ],
+            },
+          },
+          'description' => {
+            'type' => 'string',
+          },
+          'issues_url' => {
+            'type' => 'string',
+            'format' => 'uri',
+          },
+          'license' => {
+            'type' => 'string',
+          },
+          'operatingsystem_support' => {
+            'type' => 'array',
+            'items' => {
+              'properties' => {
+                'operatingsystem' => {
+                  'type' => 'string',
+                },
+                'operatingsystemrelease' => {
+                  'type' => 'array',
+                  'items' => {
+                    'type' => 'string',
+                  },
+                },
+              },
+              'required' => %w[
+                operatingsystem
+              ],
+            },
+          },
+          'name' => {
+            'type' => 'string',
+            'pattern' => '^\w+-[a-z][a-z0-9_]*$',
+          },
+          'project_page' => {
+            'type' => 'string',
+            'format' => 'uri',
+          },
+          # Undocumented but in use: https://tickets.puppetlabs.com/browse/DOCUMENT-387
+          'requirements' => {
+            'type' => 'array',
+            'items' => {
+              'properties' => {
+                'name' => {
+                  'type' => 'string',
+                },
+                'version_requirement' => {
+                  'type' => 'string',
+                },
+              },
+              'required' => %w[
+                name
+              ],
+            },
+          },
+          'source' => {
+            'type' => 'string',
+          },
+          'summary' => {
+            'type' => 'string',
+            'maxLength' => 144, # from https://forge.puppetlabs.com/razorsedge/snmp/3.3.1/scores
```

Comment:
No true, I copied this from the old check added in 6863e8f, not sure why there's no warning now. Perhaps the Forge behaviour has changed or the warning isn't displayed for old versions. I think I'll leave the check in though as the summary is meant to be "one line" according to the spec.